lib/trace.c

```diff
@@ -1108,6 +1117,10 @@ DLLEXPORT void *trace_get_packet_buffer(const libtrace_packet_t *packet,
 	assert(packet != NULL);
 	if (linktype) *linktype = trace_get_link_type(packet);
 	if (remaining) {
+                if (linktype && *linktype == TRACE_TYPE_CONTENT_INVALID) {
+                        return (void *)packet->payload;
```

Comment:
I think this should return NULL, and set remaining to 0 for safety? And if so we should always do the linktype check, regardless of linktype being supplied.

Reply:
Fixed in 86087c53c9278494e49ec101aa8fbc511256f413.